### PR TITLE
CLN: Split test_window.py

### PR DESCRIPTION
**Pull request by mroeschke**
xref #19228, #26807

- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

`test_window.py` was getting fairly unwieldily. There a couple test classes that use a non-pytest-idomatic `Base` class, so those test classes are kept together for now. I split out the rest of the test classes into separate files as they made sense.

I think the classes in `test_window.py` need further untangling before a `conftest.py` file can be made for the new directory.

Open to additional quick win optimizations.

cc @jreback

### CODE REVIEW

File: pandas/tests/window/test_dtypes.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+        data = {"sr1": sr1, "sr2": sr2, "df": df}
+
+        return data
+
+    def _create_data(self):
+        self.data = self._create_dtype_data(self.dtype)
+        self.expects = self.get_expects()
+
+    def test_dtypes(self):
```
Comment (WillAyd) [inline pandas/tests/window/test_dtypes.py]: Is this the only function being tested in this module? If so I think would make sense to try and parametrize this first before moving

File: pandas/tests/window/test_timeseries_window.py [hunk 2]
```diff
@@ -0,0 +1,692 @@
...
+        tm.assert_frame_equal(result, expected)
+
+        result = df.rolling(window="2s", min_periods=1).quantile(0.5)
+        expected = df.copy()
+        expected["B"] = [0.0, 1, 1.5, 3.0, 3.5]
+        tm.assert_frame_equal(result, expected)
+
+    def test_ragged_std(self):
+
+        df = self.ragged
```
Comment (WillAyd) [inline pandas/tests/window/test_timeseries_window.py]: Good case for parametrization here (and a few functions below)

File: pandas/tests/window/test_window.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 from collections import OrderedDict
 from datetime import datetime, timedelta
-from itertools import product
+from typing import Any
```
Comment (WillAyd) [inline pandas/tests/window/test_window.py]: Not opposed to this but note that we haven't done annotations in any other test module nor do we check it (mypy.ini excludes these) so would be OK to leave out for now as well

File: pandas/tests/window/test_window.py [hunk 4]
```diff
@@ -47,19 +39,34 @@ def raw(request):
...
+    return request.param
+
+
+@pytest.fixture(params=[True, False])
+def center(request) -> bool:
+    return request.param
+
+
+@pytest.fixture(params=[None, 1])
+def min_periods(request) -> Any:
```
Comment (WillAyd) [inline pandas/tests/window/test_window.py]: `Optional[int]` would be preferable as the return value here

**Review by WillAyd — CHANGES_REQUESTED** [review]
Thanks - this module could definitely use a little refactor.

Not opposed to this move but thinking through if we should parametrize first instead

**Comment by jreback** [comment]
@mroeschke @WillAyd let's do the straight split first, modifications later.

ping on green.

**Comment by jreback** [comment]
I think we merged something, can you merge master and ping on green.

**Comment by mroeschke** [comment]
All green @jreback

**Comment by jreback** [comment]
thanks @mroeschke

**Comment by simonjayhawkins** [comment]
it looks like this broke the ci.

**Comment by jreback** [comment]
@simonjayhawkins can u put a link up

**Comment by simonjayhawkins** [comment]
https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=14275

**Comment by pilkibun** [comment]
Strange that my PRs based on earlier commit also fails. Does azure merge the PR into master before tesing?

**Comment by simonjayhawkins** [comment]
> Does azure merge the PR into master before tesing?

it looks like it. good to know.

**Pull request merged**